content/docs/next-release-v2/architecture.md

```diff
@@ -54,4 +54,64 @@ Benefits:
 
 Downsides:
 
-* An extra layer of marshaling/unmarshaling the data.
\ No newline at end of file
+* An extra layer of marshaling/unmarshaling the data.
+
+## Jaeger binary components
+
+The Jaeger binary consists of several components, some of which are derived from OpenTelemetry and some are internal. The following diagram has the major components.
+
+![Architecture](/img/architecture-v2-binary.png)
+
+Aside from these components there are sevral other components from OpenTelemetry you can use in the config of the Jaeger binary. Here is the full list of components:
+
+### Jaeger Components
+
+* [Adaptive Sampling Processor](https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal/processors/adaptivesampling)
+
+* [Jaeger Storage Exporter](https://github.com/jaegertracing/jaeger/tree/main/cmd/jaeger/internal/extension/jaegerstorage)
```

Comment:
also `jaeger_storage` extension, which is actually the 90% of Jaeger and define all storage backends. I would put it first on the list.